File: docflow-backend/src/main/java/com/brutecx/docflow_backend/api/dto/audit/AuditExportSnapshotDTO.java
package com.brutecx.docflow_backend.api.dto.audit;

import com.brutecx.docflow_backend.domain.audit.export.AuditExportSnapshot;

import java.time.Instant;
import java.util.UUID;

public record AuditExportSnapshotDTO(
        UUID id,
        String stream,
        Instant fromTs,
        Instant toTs,
        UUID tenantId,
        String sha256DigestHex,
        long rowCount,
        Instant createdAt,
        UUID createdBy,
        String signatureB64,
        String signatureAlg,
        String keyId
) {
    public static AuditExportSnapshotDTO from(AuditExportSnapshot s) {
        return new AuditExportSnapshotDTO(
                s.getId(),
                s.getStream(),
                s.getFromTs(),
                s.getToTs(),
                s.getTenantId(),
                s.getSha256DigestHex(),
                s.getRowCount(),
                s.getCreatedAt(),
                s.getCreatedBy(),
                s.getSignatureB64(),
                s.getSignatureAlg(),
                s.getKeyId()
        );
    }
}

File: docflow-frontend/types/api/AuditExportSnapshotDTO.ts
export type AuditExportSnapshotDTO = {
  id: string; // UUID
  stream: string;

  fromTs: string; // ISO Instant
  toTs: string; // ISO Instant

  tenantId: string | null; // UUID or null (global)

  sha256DigestHex: string;
  rowCount: number;

  timestamp: string; // ISO Instant
  createdBy: string; // UUID

  signatureB64: string;
  signatureAlg: string;
  keyId: string | null;
};
